File: src/main/java/dev/kensa/render/diagram/directive/UmlTitle.java
package dev.kensa.render.diagram.directive;

import java.util.ArrayList;
import java.util.List;

import static dev.kensa.util.WrappingCollector.wrapping;
import static java.util.Arrays.asList;
import static java.util.Collections.singletonList;

public class UmlTitle implements UmlDirective {

    public static UmlTitle title(String title) {
        return new UmlTitle(title);
    }

    public static UmlTitle title(String title, String... otherLines) {
        return new UmlTitle(title, otherLines);
    }

    private final List<String> titleLines;

    private UmlTitle(String title) {
        this.titleLines = singletonList(title);
    }

    private UmlTitle(String title, String... otherLines) {
        this.titleLines = new ArrayList<>();
        this.titleLines.add(title);
        this.titleLines.addAll(asList(otherLines));
    }

    @Override
    public List<String> asUml() {
        return titleLines.stream().collect(wrapping("title", "end title"));
    }
}

File: core/src/main/kotlin/dev/kensa/render/diagram/directive/UmlTitle.kt
package dev.kensa.render.diagram.directive

class UmlTitle : UmlDirective {
    private val titleLines: List<String>

    private constructor(title: String) {
        titleLines = listOf(title)
    }

    private constructor(title: String, vararg otherLines: String) {
        titleLines = listOf(title, *otherLines)
    }

    override fun asUml(): List<String> = listOf("title") + titleLines + "end title"

    companion object {
        @JvmStatic
        fun title(title: String): UmlTitle = UmlTitle(title)

        @JvmStatic
        fun title(title: String, vararg otherLines: String): UmlTitle = UmlTitle(title, *otherLines)
    }
}
